packages/transport/protocol-v2-e2e.ts

```diff
@@ -0,0 +1,538 @@
+        handshakeInit.payload.message;
+
+    // Result of ThpCredentialRequest below
+    const knownCredentials = [
+        // {
+        //     staticPubkey: '00bf529fc8dd4662d4d1d1fa66368b8758c0b6673a1bb9d532d95ca607cbf700',
+        //     credential: undefined,
+        // },
+        {
+            trezor_static_pubkey:
+                '2991323cda7f4d5e8a1d24a79d13d93495f23e9eb8bf4348cd55d7c50bda8322',
+            credential:
+                '0a0a0a08486f73744e616d651220489054607ae8deb607303831f1257352e3262aa17e530ffea66c081890ecbd52',
+        },
+    ];
+
+    // const isKnown = protocolThp.findKnownPairingCredentials(
+    //     Buffer.from(knownCredentials[0].trezor_static_pubkey, 'hex'),
+    //     trezorEphemeralPubkey,
+    // );
+
+    const handshakeCredentials = protocolThp.handleHandshakeInitResponse({
+        hostStaticKeys,
+        hostEphemeralKeys,
+        trezorEphemeralPubkey,
+        trezorEncryptedStaticPubkey,
+        tag,
+        handshakeHash: protocolState.handshakeCredentials!.handshakeHash,
+        sendNonce: protocolState.sendNonce,
+        recvNonce: protocolState.recvNonce,
+        knownCredentials,
+    });
+
+    const { hostKey, trezorKey, trezorMaskedStaticPubkey, hostEncryptedStaticPubkey } =
+        handshakeCredentials;
+
+    // console.warn('isKnown?', Buffer.compare(isKnown, trezorMaskedStaticPubkey));
+
+    protocolState.updateHandshakeCredentials({
+        trezorEncryptedStaticPubkey,
+        hostEncryptedStaticPubkey,
+        handshakeHash: handshakeCredentials.handshakeHash,
+        trezorKey,
+        hostKey,
+    });
+
+    // protocolState.handshakeCredentials = {
+    //     trezorEncryptedStaticPubkey,
+    //     hostEncryptedStaticPubkey,
+    //     handshakeHash: handshakeHash2,
+    //     trezorKey,
+    //     hostKey,
+    // };
+
+    // if (knownCredentials.length === 1) return;
+
+    const callHandshakeComplete = await call('ThpHandshakeCompletionRequest', {
+        hostPubkey: hostEncryptedStaticPubkey,
+        noise: {
+            pairing_methods: PAIRING_METHODS,
+            host_pairing_credential: handshakeCredentials.credentials?.credential,
+        },
+        trezorMaskedStaticPubkey,
+        trezorEphemeralPubkey,
+        knownCredentials,
+    });
+    if (!callHandshakeComplete.success) {
+        throw new Error(callHandshakeComplete.message);
+    }
+
+    let isPaired = callHandshakeComplete.payload.message.state;
+
+    console.warn('callHandshakeComplete', callHandshakeComplete.payload);
+
+    if (!isPaired) {
+        const pairingReq = await call('ThpStartPairingRequest', {
+            host_name: 'HostName',
+        });
+        if (!pairingReq.success) {
+            throw new Error(pairingReq.message);
+        }
+
+        if (pairingReq.payload.message.commitment) {
+            handshakeCommitment = Buffer.from(pairingReq.payload.message.commitment, 'hex');
+        }
```

Comment:
This should be done only if code entry was a pairing method advertised in protocol parameters.